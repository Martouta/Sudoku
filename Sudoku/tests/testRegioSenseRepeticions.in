9

1

2

3

5
9

5
-1

5
0

4
0

5
0

8
0

8
9

6
3 9

5
3

8
9

3

7
1

2

7
3

2

3

8
9

8
1

8
2

0


Explicación:

Escribe el tamano de la region sin repeticion
9 //introducimos una región con 9 celdas

1 //comprobamos que el número de celdas coincide con el introducido

2 //como no hemos introducido ningún valor en ninguna celda, comprobamos que el valor de celdas rellenas sea 0

3 //vemos que toda las celdas realmente están vacías usando el método estaVacia para cada celda

5 //intentamos acceder a una posición fuera del vector de celdas (0<=pos<9) para comprobar que no nos de ningún error
9

5 //igual que en el caso anterior, comprobamos que nos de una excepción al acceder a la posición -1 del vector
-1

5 //en este caso accedemos a una posición válida del vector de celdas para comprobar que valor contienen inicialmente, pero 
	//nos devuelve un -1 ya que está vacía
0

4 //probamos las funciones de celda (set y borra) sobre una celda válida
Introduce la posicion pos de la celda
0

5 //vemos que finalmente no se ha añadido ningún valor a la celda del caso anterior
0

8 //comprobamos que un valor no válido (como lo es el 0) no está presente en la región para asegurarnos de que no haya fallos
0

8 //probamos lo mismo con el valor válido 9, y nos debería devolver un booleano false ya que no se ha añadido en ningún momento
9

6 //introducimos el valor 9 en la celda 3
3 9

5 //hacemos un get del valor de la celda 3 para comprobar que se ha introducido correctamente
3

8 //comprobamos que se ha actualizado correctamente el vector que contiene qué valores están presentes en la región con el 
	//valor 9
9

3 //vemos que realmente una de las celdas no está vacía

7 //borramos el contenido de una celda ya vacía para ver que no nos de ningún error
1

2 //vemos que el caso anterior no ha afectado al número de casillas rellenas, y comprobamos el total que llevamos

7 //en este caso eliminamos el contenido de una celda no vacía
3

2 //vemos que se ha modificado correctamente el número de celdas rellenas

3

8 //comprobamos que se ha actualizado correctamente la presencia del valor 9 en el vector de booleanos de la región, y que ahora ya no aparece
9

8
1

8
2

0