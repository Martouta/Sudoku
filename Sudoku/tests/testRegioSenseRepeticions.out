9

0

true
true
true
true
true
true
true
true
true

La posicion esta fuera del rango permitido
-2

La posicion esta fuera del rango permitido
-2

-1

9
-1

-1

El valor esta fuera del rango permitido
false

false

9

9

true

true
true
true
false
true
true
true
true
true

El valor esta fuera del rango permitido
-1

1

-1

0

true
true
true
true
true
true
true
true
true

false

false

false